### PR TITLE
Extend Sort Command

### PR DESCRIPTION
**Pull request by inezkok**
Status: DONE

Extend Sort Command to other fields:

- [x] Name
- [x] Overtime Hours
- [x] Allocated Leaves

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -32,11 +33,20 @@ public SortCommand(String attribute) {
...
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        if (attribute.toLowerCase().equals("salary")) {
-            model.updateSortedEmployeeList("salary");
-        } else if (attribute.equals("")) {
+        switch (attribute.toLowerCase()) {
+        case "":
             throw new CommandException(String.format(MESSAGE_NO_ATTR, attribute));
-        } else {
+        case "salary":
```
Comment (LordSaumya) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:39]: Future extension: could implement attribute as an enum

File: src/main/java/seedu/address/model/employee/OvertimeHours.java [hunk 2]
```diff
@@ -47,4 +47,9 @@ public int hashCode() {
         Integer thisVal = value;
         return thisVal.hashCode();
     }
+
+    @Override
+    public int compareTo(OvertimeHours other) {
+        return value - other.value;
```
Comment (LordSaumya) [inline src/main/java/seedu/address/model/employee/OvertimeHours.java:53]: Use Java's built-in `Integer.compare(value, other.value)`, it's safer.

File: src/test/java/seedu/address/logic/commands/SortCommandTest.java [hunk 3]
```diff
@@ -31,6 +33,30 @@ void execute_sortList_success() {
...
+    void execute_sortListOvertime_success() {
+        SortCommand sortCommand = new SortCommand(OVERTIME_ATTR);
+
+        String expectedMessage = String.format(SortCommand.MESSAGE_SUCCESS, OVERTIME_ATTR);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+        expectedModel.updateSortedEmployeeList(OVERTIME_ATTR);
+
+        assertCommandSuccess(sortCommand, model, expectedMessage, expectedModel);
+    }
```
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/commands/SortCommandTest.java:58]: Nice work on the new tests

File: src/main/java/seedu/address/model/employee/OvertimeHours.java (hunk 2 above)
Comment (inezkok) [reply src/main/java/seedu/address/model/employee/OvertimeHours.java]: Alright thanks I've updated this!

**Review by miljyy — APPROVED** [review]

**Pull request merged**